homeassistant/components/oasa_telematics/sensor.py

```diff
@@ -0,0 +1,195 @@
+                    })
+            params.update({
+                ATTR_ROUTE_ID: self._times[0][ATTR_ROUTE_ID],
+                ATTR_STOP_ID: self._stop_id,
+                ATTR_ATTRIBUTION: ATTRIBUTION,
+            })
+        params.update({
+            ATTR_ROUTE_NAME: self._name_data[ATTR_ROUTE_NAME],
+            ATTR_STOP_NAME: self._name_data[ATTR_STOP_NAME]
+        })
+        return {k: v for k, v in params.items() if v}
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit this state is expressed in."""
+        return 'ISO8601'
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return ICON
+
+    def update(self):
+        """Get the latest data from OASA API and update the states."""
+        self.data.update()
+        self._times = self.data.info
+        self._name_data = self.data.name_data
+        next_arrival_data = self._times[0]
+        if ATTR_NEXT_ARRIVAL in next_arrival_data:
+            self._state = next_arrival_data[ATTR_NEXT_ARRIVAL].isoformat()
+
+class OASATelematicsData():
+    """The class for handling data retrieval."""
+
+    def __init__(self, stop_id, route_id):
+        """Initialize the data object."""
+        import oasatelematics
+        self.stop_id = stop_id
+        self.route_id = route_id
+        self.info = self.empty_result()
+        self.oasa_api = oasatelematics
+        self.name_data = {ATTR_ROUTE_NAME: self.get_route_name(),
+                          ATTR_STOP_NAME: self.get_stop_name()}
+
+    def empty_result(self):
+        """Object returned when no arrivals are found."""
+        return [{ATTR_ROUTE_ID: self.route_id}]
+
+    def get_route_name(self):
+        """Get the route name from the API."""
+        try:
+            route = self.oasa_api.getRouteName(self.route_id)
+            if route:
+                return route[0].get('route_departure_eng')
+        except TypeError:
+            _LOGGER.error("Cannot get route name from OASA API")
+        return None
+
+    def get_stop_name(self):
+        """Get the stop name from the API."""
+        try:
+            name_data = self.oasa_api.getStopNameAndXY(self.stop_id)
+            if name_data:
+                return name_data[0].get('stop_descr_matrix_eng')
+        except TypeError:
+            _LOGGER.error("Cannot get stop name from OASA API")
+        return None
+
+    def update(self):
+        """Get the latest arrival data from telematics.oasa.gr API."""
+        self.info = []
+
+        results = self.oasa_api.getStopArrivals(self.stop_id)
+
+        if not results:
+            self.info = self.empty_result()
+            return
+
+        # Parse results
+        results = [r for r in results if r.get('route_code') in self.route_id]
+        current_time = dt_util.utcnow()
+
+        for result in results:
+            arrival_min = int(result.get('btime2'))
+            if arrival_min is not None:
```

Comment:
`arrival_min` can't be `None` at this point. There would have been an error at the line above where we copy to int in that case.